src/xstream.cpp

```diff
@@ -34,9 +34,28 @@ namespace xpyt
         xeus::get_interpreter().publish_stream(m_stream_name, message);
     }
 
+    void input(const std::string& prompt)
+    {
+        xeus::get_interpreter().input_request(prompt, false);
+    }
+
+    void raw_input(const std::string& prompt)
+    {
+        xeus::get_interpreter().input_request(prompt, false);
+    }
+
+    void getpass(const std::string& prompt)
+    {
+        xeus::get_interpreter().input_request(prompt, true);
```

Comment:
TODO: Await response and flush stdin/stdout buffers?

https://github.com/ipython/ipykernel/blob/118b684f75d51d72819ca90b69a245e9b02067a5/ipykernel/kernelbase.py#L875

Reply:
Closing and adding a comment in the thread.